src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -22,9 +22,8 @@ abstract class KernelGenerator extends Transformer[DenseVector[Double], DenseVec
   /* Cached KernelBlockBlock (used for training), and corresponding
    * blockIdx
    */
-  var cachedBlockBlock:Option[(DenseMatrix[Double], Seq[Int])] = None
 
-  def apply(testMat: RDD[DenseVector[Double]], blockIdxs: Seq[Int], same: Boolean = false, cache: Boolean = false): RDD[DenseVector[Double]]
+  def apply(testMat: RDD[DenseVector[Double]], blockIdxs: Seq[Int], same: Boolean = false): RDD[DenseVector[Double]]
```

Comment:
Basically we need an `Estimator` that takes in training data and produces a `Transformer` from `RDD[DenseVector[Double]]` to `KernelMatrix`, or `SimilarityMatrix`, right?